modules/distr-tracing-tempo-object-storage-setup-aws-sts-cco-install.adoc

```diff
@@ -0,0 +1,145 @@
++
+[source,yaml]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Federated": "arn:aws:iam::<aws_account_id>:oidc-provider/<oidc_provider>" # <1>
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+          "StringEquals": {
+            "<oidc_provider>:sub": [
+              "system:serviceaccount:<openshift_project_for_tempostack>:tempo-<tempostack_cr_name>" # <2>
+              "system:serviceaccount:<openshift_project_for_tempostack>:tempo-<tempostack_cr_name>-query-frontend"
+           ]
+         }
+       }
+     }
+    ]
+}
+----
+<1> The OpenID Connect (OIDC) provider that you have configured on the {product-title}.
+<2> The namespace in which you intend to create the `TempoStack` instance.
++
+[TIP]
+====
+You can also get the value for the OIDC provider by running the following command:
+----
+$ oc get authentication cluster -o json | jq -r '.spec.serviceAccountIssuer' | sed 's~http[s]*://~~g'
+----
+====
+
+. Create an AWS IAM role by attaching the created `trust.json` policy file:
++
+[source,terminal]
+----
+$ aws iam create-role \
+      --role-name "tempo-s3-access" \
+      --assume-role-policy-document "file:///tmp/trust.json" \
+      --query Role.Arn \
+      --output text
+----
+
+. Attach an AWS IAM policy to the created AWS IAM role:
++
+[source,terminal]
+----
+$ aws iam attach-role-policy \
+      --role-name "tempo-s3-access" \
+      --policy-arn "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+----
+
+. Configure the cloud provider environment for the {TempoOperator}:
++
+[source,terminal]
+----
+$ oc patch subscription <tempo_operator_sub> \ # <1>
+          -n <tempo_operator_namespace> \ # <2>
+          --type='merge' -p '{"spec": {"config": {"env": [{"name": "ROLEARN", "value": "'"<role_arn>"'"}]}}}' # <3>
+----
+<1> Operator Subscription Name.
+<2> Namespace of the {TempoOperator}.
+<3> The AWS STS requires adding the `ROLEARN` environment variable to the Operator subcription. As the `<ROLE_ARN>` value, add the Amazon Resource Name (ARN) of the AWS IAM role that you created in step 3.
+
+. In the {product-title}, create an object storage secret with keys as follows:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <name>
+stringData:
+  bucket: <s3_bucket_name>
+  region: <s3_region>
+  role_arn: <s3_role_arn>
+type: Opaque
+----
+
+. When the object storage secret is created, add the `credentialMode` parameter to the `TempoStack` custom resource:
++
+[source,yaml]
+----
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+metadata:
+  name: <name>
+  namespace: <tempo_ns>
+spec:
+  storage:
+    secret:
+      name: aws-sts
+      type: s3 # <1>
+      credentialMode: token-cco # <2>
```

Comment:
@rubenvp8510, can I add a comment that this line is `Optional`?

Reply:
For CCO is not optional, usually if you don't specify it we can infer it from secret content, but not for CCO

Reply:
ha yes, yes, if you don't want to use CCO you can configure using other methods.

Reply:
The credentialMode needs to be set to token-cco only when using CCO else we can skip the parameter or set it to token for STS without CCO.